src/Type/Php/MbSubstituteCharacterDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,107 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Php\PhpVersion;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\BooleanType;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\IntegerRangeType;
+use PHPStan\Type\NeverType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+
+use function strpos;
+
+class MbSubstituteCharacterDynamicReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	private PhpVersion $phpVersion;
+
+	public function __construct(PhpVersion $phpVersion)
+	{
+		$this->phpVersion = $phpVersion;
+	}
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'mb_substitute_character';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$phpVersion = $this->phpVersion->getVersionId();
```

Comment:
Instead of this I'd like specific methods to be added on `PhpVersion` class that asks the question you're answering here.

Reply:
Just so I'm clear on what you are saying, you'd like me to add something like `PhpVersion::isLessThan(int $versionId): bool` and other similar methods?

Reply:
Ah, I see what you mean, I just looked at `PhpVersion`.

Reply:
I mean methods like `isEmptyStringValidAliasForNoneInMbSubstituteCharacter()` etc.